src/puppetlabs/trapperkeeper/authorization/ring.clj

```diff
@@ -38,13 +38,13 @@
    certificate :- (schema/maybe X509Certificate)]
   (assoc-in request [:authorization :certificate] certificate))
 
-(schema/defn authorized-name :- schema/Str
+(schema/defn authorized-name :- (schema/maybe schema/Str)
   "Get the authorized client name from the request."
   [request :- Request]
-  (get-in request [:authorization :name] ""))
+  (get-in request [:authorization :name]))
 
 (schema/defn set-authorized-name :- Request
   "Set the authorized client name onto the request."
   [request :- Request
    name :- (schema/maybe schema/Str)]
-  (assoc-in request [:authorization :name] (str name)))
+  (assoc-in request [:authorization :name] name))
```

Comment:
Actually my comment above would be more pertinent to the getter than the setter here.